app/components/post_list/post/body/content/permalink_preview/permalink_preview.tsx

```diff
@@ -0,0 +1,214 @@
+            lineHeight: 20,
+        },
+
+        channelContext: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+        },
+        channelName: {
+            color: theme.linkColor,
+            ...typography('Body', 75, 'SemiBold'),
+        },
+    };
+});
+
+const PermalinkPreview = ({embedData, showPermalinkPreviews, author, locale, teammateNameDisplay, isOriginPostDeleted, location}: PermalinkPreviewProps) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const textStyles = getMarkdownTextStyles(theme);
+    const blockStyles = getMarkdownBlockStyles(theme);
+
+    if (!showPermalinkPreviews || isOriginPostDeleted) {
+        return null;
+    }
+
+    const {
+        post,
+        channel_display_name,
+        channel_type,
+    } = embedData;
+
+    const truncatedMessage = useMemo(() => {
+        const message = post?.message;
+
+        if (!message || typeof message !== 'string') {
+            return '';
+        }
+
+        const cleanMessage = message.trim();
+        const lines = cleanMessage.split('\n');
+        if (lines.length > MAX_PERMALINK_PREVIEW_LINES) {
+            return lines.slice(0, MAX_PERMALINK_PREVIEW_LINES).join('\n') + '...';
+        }
+
+        return cleanMessage;
+    }, [post?.message]);
+
+    const isEdited = useMemo(() => post && post.edit_at, [post]);
+
+    const authorDisplayName = useMemo(() => {
+        if (author) {
+            return displayUsername(author, locale, teammateNameDisplay);
+        }
+        return displayUsername(undefined, locale, teammateNameDisplay);
+    }, [author, locale, teammateNameDisplay]);
+
+    const channelContextText = useMemo(() => {
+        const displayName = channel_type === 'D' ? authorDisplayName : channel_display_name;
+        return `~${displayName}`;
+    }, [channel_display_name, channel_type, authorDisplayName]);
+
+    const handlePress = usePreventDoubleTap(useCallback(() => {
+        // Navigation will be implemented in Task 5
+    }, []));
+
+    if (!post) {
+        return null;
+    }
+
+    return (
+        <Pressable
+            style={({pressed}) => [
+                styles.container,
+                {opacity: pressed ? 0.8 : 1},
+            ]}
+            onPress={handlePress}
+            testID='permalink-preview-container'
+        >
+            <View style={styles.header}>
+                <ProfilePicture
+                    author={author || {id: post.user_id} as UserModel}
+                    size={32}
+                    showStatus={false}
+                />
+                <View style={styles.authorInfo}>
+                    <Text
+                        style={styles.authorName}
+                        numberOfLines={1}
+                    >
+                        {authorDisplayName}
+                    </Text>
+                    <FormattedTime
+                        timezone=''
```

Comment:
why is timezone empty here?

Reply:
Nice catch. It should use current user timestamp. Updated it.